### PR TITLE
Misuv 113: update to v3 calc model

### PR DESCRIPTION
**Pull request by scullythornton1**

### CODE REVIEW

File: app/views/estimatedTaxLiability.scala.html [hunk 1]
```diff
@@ -28,70 +28,63 @@
...
-                        BreadcrumbItem("breadcrumb-it-estimate", None, Some(messages("breadcrumb-estimated_tax_liability", s"${taxYear-1}", s"$taxYear")))
-                        )), s"EstimateFor${taxYear}Page"))
-                    }else{
-                        Some(breadcrumbHelper(Breadcrumb(Vector(
-                        BreadcrumbItem("breadcrumb-it", Some(controllers.routes.HomeController.home.url)),
-                        BreadcrumbItem("breadcrumb-estimates", Some(controllers.routes.EstimatesController.viewEstimateCalculations.url)),
-                        BreadcrumbItem("breadcrumb-it-estimate", None, Some(messages("breadcrumb-estimated_tax_liability", s"${taxYear-1}", s"$taxYear")))
-                        )), s"EstimateFor${taxYear}Page"))
-                    }
+title = messages("estimated_tax_liability.title", s"${taxYear-1}", s"$taxYear"),
```
Comment (JSimmons072019) [inline app/views/estimatedTaxLiability.scala.html]: The line format has gone

File: app/views/getLatestCalculation/estimate.scala.html [hunk 2]
```diff
@@ -28,53 +28,59 @@
...
 @main_template(
-    title = messages("estimated_tax_liability.heading", s"${model.taxYear-1}", s"${model.taxYear}"),
-    bodyClasses = None,
-    scriptElem = None,
-    contentHeader = Some(breadcrumbHelper(Breadcrumb(Vector(
-                            BreadcrumbItem("breadcrumb-it", Some(controllers.routes.HomeController.home.url)),
-                            BreadcrumbItem("breadcrumb-estimates", Some(controllers.routes.EstimatesController.viewEstimateCalculations.url)),
-                            BreadcrumbItem("breadcrumb-it-estimate", None, Some(messages("estimated_tax_liability.heading", s"${model.taxYear-1}", s"${model.taxYear}")))
-                         )), s"EstimateFor${model.taxYear}Page"))
+title = messages("estimated_tax_liability.heading", s"${model.taxYear-1}", s"${model.taxYear}"),
```
Comment (JSimmons072019) [inline app/views/getLatestCalculation/estimate.scala.html]: Line format is gone

File: it/controllers/BillsControllerISpec.scala [hunk 3]
```diff
@@ -60,23 +71,40 @@ class BillsControllerISpec extends ComponentSpecBase {
...
-        "return the correct page with tax links when calls successful" in {
+        "return the correct page with bills links when calls successful" in {
 
           And("I wiremock stub a successful Income Source Details response with Multiple Business and Property income")
           IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, multipleBusinessesAndPropertyResponse)
 
-          And("I wiremock stub a successful Get Last Estimated Tax Liability response")
-          IncomeTaxViewChangeStub.stubGetLatestCalculation(testNino, testYear, taxCalculationCrystallisedResponse)
-          IncomeTaxViewChangeStub.stubGetLatestCalculation(testNino, testYearPlusOne, taxCalculationCrystallisedResponse)
+          And("I stub a successful calculation response for 2017-18")
```
Comment (JSimmons072019) [inline it/controllers/BillsControllerISpec.scala]: Are dates that far back still relevant?

File: it/controllers/EstimatesControllerISpec.scala [hunk 4]
```diff
@@ -39,14 +42,22 @@ class EstimatesControllerISpec extends ComponentSpecBase {
           And("I wiremock stub a successful Income Source Details response with single Business and Property income")
           IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, businessAndPropertyResponse)
 
-          And("I wiremock stub a successful Get Latest Tax Calculation response")
-          IncomeTaxViewChangeStub.stubGetLatestCalculation(testNino, testYear, taxCalculationResponse)
+          And("I stub a successful calculation response for 2017-18")
```
Comment (JSimmons072019) [inline it/controllers/EstimatesControllerISpec.scala]: is a tax period two years back relevant?

File: it/controllers/HomeControllerISpec.scala [hunk 5]
```diff
@@ -42,15 +44,35 @@ class HomeControllerISpec extends ComponentSpecBase with ImplicitDateFormatter {
         IncomeTaxViewChangeStub.stubGetReportDeadlines(testSelfEmploymentId, testNino, singleObligationQuarterlyReturnModel)
         IncomeTaxViewChangeStub.stubGetReportDeadlines(testPropertyIncomeId, testNino, singleObligationOverdueModel)
         IncomeTaxViewChangeStub.stubGetReportDeadlines(testMtditid, testNino, singleObligationCrystallisationModel)
-        IncomeTaxViewChangeStub.stubGetLatestCalculation(testNino, testYear, taxCalculationResponseCrystallised)
-        IncomeTaxViewChangeStub.stubGetLatestCalculation(testNino, testYearPlusOne, taxCalculationResponseCrystallised)
+        And("I stub a successful calculation response for 2017-18")
```
Comment (JSimmons072019) [inline it/controllers/HomeControllerISpec.scala]: Is this tax year relevant

File: test/services/CalculationServiceSpec.scala [hunk 6]
```diff
@@ -40,60 +39,32 @@ class CalculationServiceSpec extends TestSupport with MockIncomeTaxViewChangeCon
...
     "when the  Calculation Data api feature switch is disabled" should {
 
       "passed an ordered list of years" should {
 
         "for a list of Estimates" should {
 
           "return a list of CalculationResponseModelWithYear models" in new CalculationDataApiDisabled {
-            setUpLatestCalculationResponse(testNino, testYear)(testCalcModelEstimate)
-            setUpLatestCalculationResponse(testNino, testYearPlusOne)(testCalcModelEstimate)
+            mockGetLatestCalculationId(testNino, "2017-18")(Right("testIdOne"))
```
Comment (JSimmons072019) [inline test/services/CalculationServiceSpec.scala]: relevant dates?

File: it/controllers/BillsControllerISpec.scala (hunk 3 above)
Comment (scullythornton1) [reply it/controllers/BillsControllerISpec.scala]: it does not matter which year we use, as long as the service brings back the same tax year as the stub one.

File: it/controllers/EstimatesControllerISpec.scala (hunk 4 above)
Comment (scullythornton1) [reply it/controllers/EstimatesControllerISpec.scala]: in estimate and bills, the user can view the previous tax year calculation detail, so to stub a previous tax year is reasonable for verifying the code has brought back the correct tax year.

File: it/controllers/HomeControllerISpec.scala (hunk 5 above)
Comment (scullythornton1) [reply it/controllers/HomeControllerISpec.scala]: the same reasons as above

File: test/services/CalculationServiceSpec.scala (hunk 6 above)
Comment (scullythornton1) [reply test/services/CalculationServiceSpec.scala]: the same reason as above

**Pull request merged**